src/App.tsx

```diff
@@ -1,73 +1,84 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { Dropdown } from './components/Dropdown';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  debounce?: number;
+  onSelected?: (person: Person) => void;
+};
 
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+export const App: React.FC<Props> = ({ debounce = 300, onSelected }) => {
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [filteredPeople, setFilteredPeople] =
+    useState<Person[]>(peopleFromServer);
 
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+  const { name, born, died } = selectedPerson || {};
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+  const handleSelected = useCallback(
+    (person: Person) => {
+      setSelectedPerson(person);
+      if (onSelected) {
+        onSelected(person);
+      }
+    },
+    [onSelected],
+  );
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+  const handleQuery = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+    },
+    [],
+  );
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setFilteredPeople(
+        peopleFromServer.filter(person =>
+          person.name.toLowerCase().includes(query.toLowerCase()),
+        ),
+      );
+    }, debounce);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+    return () => clearTimeout(timer);
+  }, [query, debounce]);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+  useEffect(() => {
+    if (query !== '' && selectedPerson) {
```

Comment:
```suggestion
    if (!!query && selectedPerson) {
```

can be a bit simplify

Reply:
дякую, забуваю про цей варіант !!
Виправила